madsim/src/sim/task/mod.rs

```diff
@@ -623,7 +630,7 @@ impl Spawner {
         F::Output: 'static,
     {
         if self.info.killed.load(Ordering::Relaxed) {
-            panic!("spawning task on a killed node");
+            tracing::warn!("spawning task on a killed node, the task will never run");
```

Comment:
Also relax this check so that `spawn` during drop by killing a node won't panic.